umbraco/Umbraco-CMS
# V10: Migrate member properties to columns on the member table

This migrates most of the default Umbraco properties on the member types to be saved directly on the member row in the `cmsMember` table, these properties are:

* umbracoMemberFailedPasswordAttempts
* umbracoMemberApproved
* umbracoMemberLockedOut
* umbracoMemberLastLockoutDate
* umbracoMemberLastLogin
* umbracoMemberLastPasswordChangeDate

These properties have also been moved in the backoffice so they're now in the Member content app instead of content. 

A thing worth mentioning here is that by default all these properties were marked as sensitive, so, for now, these are hardcoded as sensitive in the member content app, we may want to consider a way to configure this at some point. 

Additionally, the date properties are now nullable since they're initialized to null (for instance when you've never logged in), instead of them being `DateTime.MinValue`, these properties are:
* LastLoginDate
* LastPasswordChangeDate
* LastLockoutDate

Lastly, I've re-used the user translations for the properties instead of hardcoded labels, and I've added the `stateApproved` translation because that was the only missing one. 

This also fixes #11931 for V10

## Breaking changes

This PR is breaking because you can no longer access the properties listed above through the `IMember.Properties`, collection, you must now access them through their specific properties I.E.  `IMember.IsLockedOut`. 

Additionally, when previously you resolved a member as published content, all the default properties would be there twice, for instance, `IsLockedOut` would be there both as a property with the alias `umbracoMemberLockedOut` and with the alias `IsLockedOut`, now it'll only be there once, with the alias being the name of the property, so `IsLockedOut` in this instance.

Lastly the nullable dates on a user, I.E. `LastLoginLate` will now be null instead of `DateTime.MinValue` when getting a user with the user service.

## Testing

⚠️ Front end files have updated, remember to run `npm run build` before testing ⚠️

In general members and users should work as before, some things that might be a good idea to test:

* Test the migration, with custom member types and member groups
* Ensure you can CRUD members both in the frontend and backoffice
* Ensure that properties like `LastLoginDate` updates correctly
